core/src/rpc_service.rs

```diff
@@ -239,6 +253,26 @@ impl RequestMiddleware for RpcRequestMiddleware {
     }
 }
 
+fn process_rest(bank_forks: &Arc<RwLock<BankForks>>, path: &str) -> Option<String> {
+    match path {
```

Comment:
I initially plumbed this to `JsonRpcRequestProcessor::get_supply()` to avoid the duplication, but that caused a circular reference.   The duplication is pretty minimal and simple so not too worried.